### ISSUE TITLE
Mark class as problematic whenever a method argument miss types

### ISSUE THREAD

**Issue by d8vjork** [DEVSENSE/phptools-docs#234]
### Actual behaviour

Here I've a visual example:

<img width="1092" alt="image" src="https://user-images.githubusercontent.com/2331052/210538039-69d760bb-a4ea-4f12-a75d-2355e3195924.png">

In this example I'm implementing `Illuminate\Contracts\Database\Eloquent\CastsAttributes` interface which enforces me to use proper models instead of my own `Statusable` interfaced models, so this is the error you could find at PHP runtime (in this case, at tests):

```
OpenSoutheners\LaravelModelStatus\Casts\Status::get(

OpenSoutheners\LaravelModelStatus\Statusable $model, string $key, OpenSoutheners\LaravelModelStatus\ModelStatus $value, array $attributes

):

string must be compatible with Illuminate\Contracts\Database\Eloquent\CastsAttributes::get(

$model, string $key, $value, array $attributes

) in /Users/d8vjork/Projects/open-southeners/laravel-model-status/src/Casts/Status.php on line 16
```

### Expected behaviour

To warn about the error as it should normally do (at the `class X implements Y`)

**Comment by d8vjork** [comment]
Ok I confused, right now what this does is notify when a method is missing from the interface, only that

So this should be more a feature request

**Comment by jakubmisek** [comment]
that's a good diagnostic ... PHP has a few rules on how the types in subclasses can be changed; we should be able to analyze that

**Comment by parker-chung** [comment]
Hello @jakubmisek  
I'm just wondering if there's any plan to add this?
Thank you in advance.

**Comment by jakubmisek** [comment]
@parker-chung thank you for reminding us,

yes, we are planning to add this diagnostic, soon :)

**Comment by jakubmisek** [comment]
We have released update `1.45.15061` with the first bunch of override diagnostics (not checking the type inheritence yet).

Next update(s) will introduce the type hint overriding diagnostics.

**Comment by d8vjork** [comment]
Awesome! This can count as a proof :)

<img width="976" alt="image" src="https://github.com/DEVSENSE/phptools-docs/assets/2331052/e099cea3-19c4-4a47-a1f2-7e01d6b99f46">
